src/js/popup.js

```diff
@@ -67,27 +67,30 @@ function loadTemplateEditor (openTemplate = null) {
         $('#addDefaultDropdown').empty();
         // remove all domains, so that the whole list can be re-added in proper order.
         $('.custom-domain-collection').remove();
+        // remove all input ids, so that the whole list can be re-added in proper order.
+        $('.custom-inputID-collection').remove();
 
-        if (templates[StorageID].templates) {
-            templates = templates[StorageID].templates;
+        templates = templates[StorageID].templates;
 
-            $('#templateEditorTitle').text('Templates:');
-
-            // Sort Alphabetically except with DEFAULT TEMPLATE at the top.
-            var templatesArray = sortTemplates(templates);
-
-            // Send a message to sandbox.html to build the collapsible template editor
-            // Once the template is compiled, a 'message' event will be sent to this window with the html
-            var sandboxIFrame = document.getElementById('sandbox_window');
-            sandboxIFrame.contentWindow.postMessage({
-                command: 'renderTemplates',
-                context: { templates: templatesArray },
-                openTemplate: openTemplate
-            }, '*');
+        var tempLoaded;
+        if (jQuery.isEmptyObject(templates)) {
+            tempLoaded = false;
         } else {
-            $('#templateEditorTitle').text('No templates are currently loaded');
+            tempLoaded = true;
```

Comment:
You should be able to do all this logic in the template. Check if `templates` is empty or not. i.e we shouldn't need the extra variable `tempLoaded`

Reply:
Yes that makes perfect sense and simplifies the code a lot, thanks for pointing it out I appreciate it